### PR TITLE
Added custom rule to update all_activity_complete_date to today

### PR DESCRIPTION
**Pull request by orangejenny**
## Product Description
https://dimagi-dev.atlassian.net/browse/USH-1696

### Safety story
Custom code, will only affect USH projects that use the rule.

### Automated test coverage

Tests in PR.

### QA Plan

Not requesting QA. I'll ask delivery to run it in a test space before turning it on for a production project.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: custom/covid/rules/custom_actions.py [hunk 1]
```diff
@@ -4,13 +4,41 @@
...
 from corehq.apps.es import filters
 
 
+def set_all_activity_complete_date_to_today(case, rule):
+    """
+    For any case matching the criteria, set the all_activity_complete_date property
+    to today's date, in YYYY-MM-DD format, based on UTC.
+    """
+    domain_obj = Domain.get_by_name(case.domain)
+    today = datetime.now(domain_obj.get_default_timezone()).strftime(ISO_DATE_FORMAT)
```
Comment (snopoke) [inline custom/covid/rules/custom_actions.py:25]: The docstring says 'based on UTC' but here you're using the timezone of the domain.

**Review by snopoke — COMMENTED** [review]
I'm curious how this interacts with the time the task rule get's run and what a project means by 'today'. I could imagine that a project would want it to be 'the day the task was meant to run' which would always be the UTC day and not necessarily the day in the domains TZ at the instant the rule was run.

I think the tasks kick off at midnight so:

- if the domain TZ is behind UTC then 'today' could be the day prior to when the task ran in UTC e.g. rule runs at 01:00 and TZ = UTC-5

- if the domain TZ is ahead of UTC then 'today' will almost always be the same as the day in the TZ except in rare circumstances where the offset is large or the rules are delayed in which case 'today' could be 'tomorrow'.  e.g. rule runs at 18:00 and TZ = UTC+8

File: custom/covid/rules/custom_actions.py (hunk 1 above)
Comment (orangejenny) [reply custom/covid/rules/custom_actions.py:25]: oops, forgot to update that

File: custom/covid/rules/custom_actions.py [hunk 2]
```diff
@@ -4,13 +4,41 @@
...
 from corehq.apps.es.cases import case_type
 from corehq.apps.data_interfaces.models import CaseRuleActionResult, AUTO_UPDATE_XMLNS
 from corehq.apps.hqcase.utils import update_case
 from corehq.apps.es import filters
 
 
+def set_all_activity_complete_date_to_today(case, rule):
+    """
+    For any case matching the criteria, set the all_activity_complete_date property
+    to today's date, in YYYY-MM-DD format, based on UTC.
```
Comment (orangejenny) [inline custom/covid/rules/custom_actions.py:22]:
```suggestion
    to today's date, in YYYY-MM-DD format, based on the domain's default time zone.
```

**Comment by orangejenny** [comment]
@snopoke Yeah, that's a good point. I was thinking the rules run at 7 pm, but a) you're right it's midnight, and b) even if it were 7, it's still be questionable. I'll check with the project team.

**Review by esoergel — APPROVED** [review]

**Comment by orangejenny** [comment]
@snopoke The project team confirmed they're at peace with the date being the date the rule runs, even though that's the day after the case was actually last updated.

**Pull request merged**